x/stake/validator.go

```diff
@@ -0,0 +1,242 @@
+package stake
+
+import (
+	"bytes"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/wire"
+	abci "github.com/tendermint/abci/types"
+	crypto "github.com/tendermint/go-crypto"
+)
+
+// Validator defines the total amount of bond shares and their exchange rate to
+// coins. Accumulation of interest is modelled as an in increase in the
+// exchange rate, and slashing as a decrease.  When coins are delegated to this
+// validator, the validator is credited with a Delegation whose number of
+// bond shares is based on the amount of coins delegated divided by the current
+// exchange rate. Voting power can be calculated as total bonds multiplied by
+// exchange rate.
+type Validator struct {
+	Status  sdk.BondStatus `json:"status"`  // bonded status
+	Address sdk.Address    `json:"address"` // sender of BondTx - UnbondTx returns here
+	PubKey  crypto.PubKey  `json:"pub_key"` // pubkey of validator
+
+	PShares         PoolShares `json:"pool_shares"`      // total shares for tokens held in the pool
```

Comment:
PoolShares

Reply:
not descriptive enough - do you mean to say you don't like PShares?